client/src/views/SearchResults.tsx

```diff
@@ -0,0 +1,253 @@
+  },
+  contentWrapper: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+  },
+  leftPanel: {
+    borderRight: '1px solid grey',
+    width: '20%',
+  },
+  createBar: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+  rightPanel: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    padding: '10px 5%',
+    width: '80%',
+  },
+  makeAPost: {
+    fontWeight: 'bold',
+    textDecoration: 'none',
+    color: 'black',
+  },
+}));
+
+function SearchResults(): JSX.Element {
+  const classes = useStyles();
+  const location = useLocation<{ category: '' }>();
+  //implement below line to input filters at render
+  const querySearchCategory = location.state.category || '';
+  const querySearchText = queryString.parse(location.search).search;
+  const [searchCategory, setSearchCategory] = React.useState<string>(
+    (querySearchCategory as string) || '',
+  );
+  const [searchText, setSearchText] = React.useState<string>((querySearchText as string) || '');
+  const [inputText, setInputText] = React.useState<string>('');
+  const [selectedFilters, setSelectedFilters] = React.useState<{ [key: string]: boolean }>({});
+
+  const handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedFilters({
+      ...selectedFilters,
+      [event.target.value]: event.target.checked,
+    });
+  };
+
+  const [offers, setOffers] = React.useState<Asset[]>([]);
+  const [needs, setNeeds] = React.useState<Asset[]>([]);
+  const [orgs, setOrgs] = React.useState<Asset[]>([]);
+  const [volunteer, setVolunteer] = React.useState<Asset[]>([]);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setSearchCategory(event.target.value);
+  };
+
+  const handleSearch = () => {
+    setSearchText(inputText);
+    setInputText('');
+  };
+
+  const handleInput = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  React.useEffect(() => {
+    // fetch assets with searchText
+    console.log(`Search this: ${searchCategory}`);
+    if (searchCategory === 'Volunteer') {
+      //TODO: Change API to /volunteer
+      fetch(`${APP_API_BASE_URL}/organizations`)
+        .then((resp) => resp.json())
+        .then((data: Asset[]) => {
+          setVolunteer(data);
+        });
+    } else if (searchCategory === 'Offers' || searchCategory === 'Needs') {
+      fetch(
+        `${APP_API_BASE_URL}/assets?type=${searchCategory === 'Needs' ? 'donation' : 'request'}${
+          searchText ? `&title=${searchText}` : ''
+        }`,
+      )
+        .then((resp) => resp.json())
+        .then((data: Asset[]) => {
+          if (searchCategory === 'Needs') {
+            setNeeds(data);
+          } else {
+            setOffers(data);
+          }
+        });
+    } else if (searchCategory === 'Nonprofits') {
+      fetch(`${APP_API_BASE_URL}/organizations`)
```

Comment:
so here's where you would add `${APP_API_BASE_URL}/organizations?search=${searchText}` to utilize the BE feature i added in that PR

Reply:
Got it! Linking the branch here: https://github.com/Nonprofit-Exchange-Hub/web-app/pull/229#issuecomment-1289162380